iModelJsNodeAddon/JsCloudSqlite.cpp

```diff
@@ -191,8 +191,20 @@ struct JsCloudContainer : CloudContainer, Napi::ObjectWrap<JsCloudContainer> {
 
     void RequireWriteLock() {
         RequireConnected();
-        if (!m_writeLockHeld)
-            BeNapi::ThrowJsException(Env(), Utf8PrintfString("container [%s] is not locked for write access", m_containerId.c_str()).c_str());
+        BeJsDocument lockedBy;
+        ReadWriteLock(lockedBy);
+        auto lockedByGuid = lockedBy[JSON_NAME(guid)].asString();
+        // check if it's the same guid  
+        if (lockedByGuid != m_cache->m_guid) {
+            // another user grabbed the write lock after the current user's write lock expiration time, disable current user from operating
+            BeNapi::ThrowJsException(Env(), Utf8PrintfString("Container [%s] is currently locked by another user.", m_containerId.c_str()).c_str());
+        } else {
+            // Refresh the write lock for the current user no matter if the write lock expires
+            // 1. If the write lock expires, refresh it for the current user to operate further actions
+            // 2. If not, also refresh it in case the current user's write lock is about to expire when they call operations that require write lock to be present, and somehow expires during the operations

```

Comment:
Could you also potentially write a test on itwinjs side that asserts the write lock gets refreshed for a user? 

You might have to read bcv_kv similar to how you modified it in azuriteTest in order to achieve this.

Reply:
added in [commit ee5297e0e535456bd7b00797255309cbbeca0734](https://github.com/iTwin/itwinjs-core/pull/7000/commits/ee5297e0e535456bd7b00797255309cbbeca0734)